### PR TITLE
Config parsing

### PR DESCRIPTION
**Pull request by csware**
This PR fixes issues #6089 and #6088

### CODE REVIEW

File: tests/config/read.c [hunk 1]
```diff
@@ -219,6 +233,18 @@ void test_config_read__multiline_multiple_quoted_comment_chars(void)
...
 }
 
+void test_config_read__multiline_multiple_quoted_quote_at_beginning_of_line(void)
+{
+	git_config* cfg;
+	cl_git_pass(git_config_open_ondisk(&cfg, cl_fixture("config/config22")));
+	cl_git_pass(git_config_get_string_buf(&buf, cfg, "alias.m"));
+	cl_assert_equal_s("cmd ;; ;; bar", buf.ptr);
+	git_buf_dispose(&buf);
+	cl_git_pass(git_config_get_string_buf(&buf, cfg, "alias.m2"));
```
Comment (ethomson) [inline tests/config/read.c:243]: CI is failing, it looks like because of a memory leak.  Possibly here?

File: tests/config/write.c [hunk 2]
```diff
@@ -743,3 +745,19 @@ void test_config_write__preserve_case(void)
...
+{
+	git_config* cfg;
+	git_buf buf = GIT_BUF_INIT;
+
+	cl_git_pass(git_config_open_ondisk(&cfg, "config22"));
+	cl_git_pass(git_config_get_string_buf(&buf, cfg, "alias.m"));
+	cl_assert_equal_s("cmd ;; ;; bar", buf.ptr);
+	cl_git_pass(git_config_set_string(cfg, "sOMe.ThInG", "foo"));
+	git_buf_dispose(&buf);
+	cl_git_pass(git_config_get_string_buf(&buf, cfg, "alias.m"));
```
Comment (ethomson) [inline tests/config/write.c:759]: Or here?

File: tests/config/read.c [hunk 3]
```diff
@@ -209,6 +209,20 @@ void test_config_read__symbol_headers(void)
...
+	cl_git_pass(git_config_get_string_buf(&buf, cfg, "sec.[subsec]/child.parent"));
+	cl_assert_equal_s("grand", buf.ptr);
+	git_buf_dispose(&buf);
+	cl_git_pass(git_config_get_string_buf(&buf, cfg, "sec2.[subsec2]/child2.type"));
+	cl_assert_equal_s("dvcs", buf.ptr);
+	git_buf_dispose(&buf);
+	cl_git_pass(git_config_get_string_buf(&buf, cfg, "sec3.escape\"quote.vcs"));
+	cl_assert_equal_s("git", buf.ptr);
+	git_buf_dispose(&buf);
+	cl_git_pass(git_config_get_string_buf(&buf, cfg, "sec4.escaping\\slash.lib"));
```
Comment (ethomson) [inline tests/config/read.c:224]: Possibly this is the culprit...

**Comment by csware** [comment]
@ethomson Should be fixed now.

**Comment by ethomson** [comment]
Thanks for the fix!

**Pull request merged**